it/controllers/AtsMergePageControllerItSpec.scala

```diff
@@ -0,0 +1,319 @@
+
+    val authResponse =
+      s"""
+         |{
+         |    "confidenceLevel": 200,
+         |    "nino": "$generatedNino",
+         |    "saUtr": "$generatedSaUtr",
+         |    "name": {
+         |        "name": "John",
+         |        "lastName": "Smith"
+         |    },
+         |    "loginTimes": {
+         |        "currentLogin": "2021-06-07T10:52:02.594Z",
+         |        "previousLogin": null
+         |    },
+         |    "optionalCredentials": {
+         |        "providerId": "4911434741952698",
+         |        "providerType": "GovernmentGateway"
+         |    },
+         |    "authProviderId": {
+         |        "ggCredId": "xyz"
+         |    },
+         |    "externalId": "testExternalId",
+         |    "allEnrolments": [{
+         |        "key": "IR-SA-AGENT",
+         |        "identifiers": [
+         |          {
+         |            "key": "IRAgentReference",
+         |            "value": "uar"
+         |          }],
+         |        "state": "Activated"
+         |     }]
+         |}
+         |""".stripMargin
+
+    server.stubFor(post(urlEqualTo("/auth/authorise"))
+      .willReturn(ok(authResponse)))
+  }
+
+  when(mockDataCacheConnector.storeAgentToken(any[String])(any[HeaderCarrier], any[ExecutionContext]))
+    .thenReturn(Future.successful("token"))
+
+  when(mockDataCacheConnector.fetchAndGetAtsListForSession(any[HeaderCarrier]))
+    .thenReturn(Future.successful(Some(atsListData)))
+
+  when(mockDataCacheConnector.getAgentToken(any[HeaderCarrier], any()))
+    .thenReturn(Future.successful(Some(agentTokenMock)))
+
+  when(mockDataCacheConnector.storeAtsListForSession(any())(any(), any()))
+    .thenReturn(Future.successful(Some(atsListData)))
+
+  "/income-before-tax" must {
+
+    lazy val url = s"/annual-tax-summary/paye/main?ref=PORTAL&id=$agentToken"
+
+    lazy val backendUrlSa = s"/taxs/$generatedSaUtr/ats-list"
+
+    lazy val backendUrlPaye = s"/taxs/$generatedNino/${appConfig.taxYear - appConfig.maxTaxYearsTobeDisplayed}/${appConfig.taxYear}/paye-ats-data"
+
+    "return an OK response with appropriate query parameters for Agent when data is retrieved from backend for both atsList and payeData" in {
+
+      server.stubFor(
+        get(urlEqualTo(backendUrlSa))
+          .willReturn(ok(FileHelper.loadFile("./it/resources/atsList.json")))
+      )
+
+      server.stubFor(
+        get(urlEqualTo(backendUrlPaye))
+          .willReturn(ok(FileHelper.loadFile("./it/resources/payeData.json")))
+      )
+
+      val request = FakeRequest(GET, url)
+
+      val result = route(fakeApplication(), request)
+
+      result.map(status) mustBe Some(OK)
+
+      request.getQueryString(Globals.TAXS_USER_TYPE_QUERY_PARAMETER) mustBe Some("PORTAL")
+
+      request.getQueryString(Globals.TAXS_AGENT_TOKEN_ID).isDefined mustBe true
+    }
+
+    "return an OK response without query parameters when data is retrieved from backend for both atsList and payeData" in {
+
```

Comment:
Is this the empty token test? If so can the name of the test say so?

Reply:
Done